pkg/webhooks/updaterequest/generator.go

```diff
@@ -66,7 +68,21 @@ func (g *generator) tryApplyResource(ctx context.Context, urSpec kyvernov1beta1.
 	if urSpec.GetRequestType() == kyvernov1beta1.Mutate {
 		queryLabels = common.MutateLabelsSet(urSpec.Policy, urSpec.GetResource())
 	} else if urSpec.GetRequestType() == kyvernov1beta1.Generate {
-		queryLabels = common.GenerateLabelsSet(urSpec.Policy, urSpec.GetResource())
+		resource := &unstructured.Unstructured{}
+		admissionRequest := urSpec.Context.AdmissionRequestInfo.AdmissionRequest
+		if admissionRequest != nil {
+			var err error
+			if admissionRequest.Operation == "DELETE" {
+				err = json.Unmarshal(urSpec.Context.AdmissionRequestInfo.AdmissionRequest.OldObject.Raw, &resource)
+			} else {
+				err = json.Unmarshal(urSpec.Context.AdmissionRequestInfo.AdmissionRequest.Object.Raw, &resource)
+			}
+			if err != nil {
+				logger.Error(err, "failed to unmarshal admission request object")
+				return err
+			}
+		}
+		queryLabels = common.GenerateLabelsSet(urSpec.Policy, urSpec.GetResource(), string(resource.GetUID()))
```

Comment:
@MarcelMue According to my current changes, if `admissionRequest` is `nil` `string(resource.GetUID())` will become `""`.
If the resource UID can't be retrieved through `urSpec.Context` here, then I believe we'd have to pass it down through the `Apply()` function to `tryApplyResource()` which doesn't seem like the right way to do it.

Any suggestions as to how the resource UID could be retrieved properly?

Reply:
If we introduced this method `GetUID()` in the `Object` interface, we won't need these changes.